tests/lib/deep.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+var expect = require('chai').expect,
+    deepFreeze = require('../../lib/deep');
+
+function assertImmutableObject(object, modifier) {
+    var catchedErr = null;
+    try {
+        modifier(object);
+    } catch (err) {
+        catchedErr = err;
+    }
+    expect(catchedErr).to.be.an['instanceof'](Error);
+    expect(catchedErr.message).to.include('object is not extensible');
+}
+
+describe('deep freeze', function () {
+    it('should be immutable', function () {
+        deepFreeze(Buffer);
+        assertImmutableObject(Buffer, function (obj) {
+            obj.x = 5;
+        });
+        assertImmutableObject(Buffer, function (obj) {
+            obj.prototype.z = 3;
+        });
+    });
+});
```

Comment:
copied and rewrited

- https://www.npmjs.com/package/deep-freeze?activeTab=explore
- https://www.npmjs.com/package/tap

<img width="1117" alt="Screen Shot 2022-12-23 at 7 12 10 PM" src="https://user-images.githubusercontent.com/8896191/209326286-e8165101-ba7a-46f7-8751-b002325518d3.png">